src/components/shared/Popover.js

```diff
@@ -15,16 +17,18 @@ class Popover extends Component {
 
   componentDidMount() {
     const { type } = this.props;
-    const { left, top, dir } = type == "popover"
+    const { left, top, dir, targetMid } = type == "popover"
       ? this.getPopoverCoords()
       : this.getTooltipCoords();
 
     // eslint-disable-next-line react/no-did-mount-set-state
-    this.setState({ left, top, dir });
+    this.setState({ left, top, dir, targetMid });
   }
 
   getPopoverCoords() {
```

Comment:
Yeah, current implementation is hard to understand, it'll be more readable if we split the `getPopoverCoords()` function.